### PR TITLE
[Rust] Upgrade to Rust 1.71.0

### PR DESCRIPTION
**Pull request by JoshLind**
### Description
This PR updates the `aptos-core` repository to use the [latest version](https://github.com/rust-lang/rust/blob/master/RELEASES.md) of the rust toolchain.

### Test Plan
Existing test infrastructure.

### CODE REVIEW

**Review by brianolson — APPROVED** [review]

File: execution/executor-benchmark/src/lib.rs [hunk 1]
```diff
@@ -487,6 +487,49 @@ fn add_accounts_impl<V>(
     );
 }
 
+struct GasMesurement {
```
Comment (gregnazario) [inline execution/executor-benchmark/src/lib.rs:490]: I know this probably isn't you but it's spelled wrong

**Review by gregnazario — COMMENTED** [review]
Can we test this build on Windows?

**Comment by 0xjinn** [comment]
I know we just upgraded to `1.7` recently. I wonder what is the need to get to 1.71?

Also, we will most likely need to upgrade CLI homebrew formula to match this new version `1.71` - https://github.com/Homebrew/homebrew-core/blob/master/Formula/aptos.rb

I can open a PR in homebrew if needed 😄

File: execution/executor-benchmark/src/lib.rs (hunk 1 above)
Comment (JoshLind) [reply execution/executor-benchmark/src/lib.rs:490]: Yeah, this wasn't me. I didn't want to litter the PR with other fixes, hence why I've left it as is 😄

**Comment by JoshLind** [comment]
> I wonder what is the need to get to 1.71?

@0xjinn, we're hoping that the upgrade will help address some of the spurious compiler panics we've seen in relation to MIR (e.g., https://github.com/rust-lang/rust/issues/112637).

> I can open a PR in homebrew if needed 😄

Thanks @0xjinn 😄 Once this lands, we can also update the CLI docs (if we update the CLI like we did last time?)

**Comment by JoshLind** [comment]
> Can we test this build on Windows?

@gregnazario, do we have a process for this? e.g., a CI/CD job/label?

**Review by ibalajiarun — APPROVED** [review]
Docker builder changes LGTM

**Pull request merged**